### PR TITLE
[MultiDB]: throwing exception when database_config.json doesn't exist

### PR DESCRIPTION
**Pull request by dzhangalibaba**
* throwing exception if config file is missing
* using interface .at() to get map entry

Signed-off-by: Dong Zhang d.zhang@alibaba-inc.com

### CODE REVIEW

**Comment by dzhangalibaba** [comment]
@kcudnik @qiluo The exception part.

@qiluo-msft  BTW , we need to make a agreement on whether adding db config file into build docker by default to make progress on next step. Kamil and I thought it is OK.

File: common/dbconnector.cpp [hunk 1]
```diff
@@ -53,42 +53,44 @@ void SonicDBConfig::initialize(const string &file)
         {
             throw runtime_error("Sonic database config file syntax error >> " + string(e.what()));
         }
+    } else {
```
Comment (qiluo-msft) [inline common/dbconnector.cpp]: >} else { [](start = 4, length = 8)

follow code format nearby

File: common/dbconnector.cpp [hunk 2]
```diff
@@ -53,42 +53,44 @@ void SonicDBConfig::initialize(const string &file)
...
+        throw runtime_error("Sonic database config file doesn't exist at " + file);
     }
 }
 
 string SonicDBConfig::getDbInst(const string &dbName)
```
Comment (qiluo-msft) [inline common/dbconnector.cpp]: >getDbInst [](start = 22, length = 9)

Add a unit test case for non existing dbName

File: common/dbconnector.cpp [hunk 3]
```diff
@@ -53,42 +53,44 @@ void SonicDBConfig::initialize(const string &file)
...
 {
     if (!m_init)
         initialize();
-    return m_db_info[dbName].first;
+    return m_db_info.at(dbName).first;
 }
 
 int SonicDBConfig::getDbId(const string &dbName)
```
Comment (qiluo-msft) [inline common/dbconnector.cpp]: >getDbId [](start = 19, length = 7)

Add a unit test case for non existing dbName

**Comment by kcudnik** [comment]
for each throw can you also add swss_log_error message to make mark in syslog if on someplace someone will forget to catch/log catched exception?

File: common/dbconnector.cpp (hunk 1 above)
Comment (dzhangalibaba) [reply common/dbconnector.cpp]: DONE

File: common/dbconnector.cpp (hunk 2 above)
Comment (dzhangalibaba) [reply common/dbconnector.cpp]: DONE. Added all five API test cases.

File: common/dbconnector.cpp (hunk 3 above)
Comment (dzhangalibaba) [reply common/dbconnector.cpp]: DONE. Added all five API test cases.

**Comment by dzhangalibaba** [comment]
Capture Logs When exception Thrown
```
Dec  4 08:08:55.621594 ASW-7005 DEBUG syncd#syncd: :> syncd_main: enter
Dec  4 08:08:55.621837 ASW-7005 ERR syncd#syncd: :- initialize: Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json
Dec  4 08:08:55.623204 ASW-7005 INFO syncd#supervisord: syncd terminate called after throwing an instance of 'std::runtime_error'#015
Dec  4 08:08:55.623289 ASW-7005 INFO syncd#supervisord: syncd   what():  Sonic database config file doesn't exist at /var/run/redis/sonic-db/database_config.json#015
```

UT tests:
```
[ RUN      ] DBConnector.multi_db_test
Default : isInit = 0
INIT: loading nonexist db config file
INIT : load local db config file, isInit = 1
INIT: loading local config file again
GET: invalid dbname input for getDbInst()
GET: invalid dbname input for getDbId()
GET: invalid dbname input for getDbSock()
GET: invalid dbname input for getDbHostname()
GET: invalid dbname input for getDbPort()
testing APPL_DB
redis#0#/var/run/redis/redis.sock#127.0.0.1#6379
testing ASIC_DB
redis1#1#/var/run/redis/redis1.sock#127.0.0.1#6380
testing CONFIG_DB
redis2#4#/var/run/redis/redis2.sock#127.0.0.1#6381
testing COUNTERS_DB
redis1#2#/var/run/redis/redis1.sock#127.0.0.1#6380
testing FLEX_COUNTER_DB
redis3#5#/var/run/redis/redis3.sock#127.0.0.1#6382
testing LOGLEVEL_DB
redis2#3#/var/run/redis/redis2.sock#127.0.0.1#6381
testing PFC_WD_DB
redis3#5#/var/run/redis/redis3.sock#127.0.0.1#6382
testing SNMP_OVERLAY_DB
redis4#7#/var/run/redis/redis4.sock#127.0.0.1#6383
testing STATE_DB
redis4#6#/var/run/redis/redis4.sock#127.0.0.1#6383
[       OK ] DBConnector.multi_db_test (1 ms)
```

**Comment by dzhangalibaba** [comment]
> for each throw can you also add swss_log_error message to make mark in syslog if on someplace someone will forget to catch/log catched exception?

> for each throw can you also add swss_log_error message to make mark in syslog if on someplace someone will forget to catch/log catched exception?

Yes. Added.

File: tests/redis_multi_db_ut.cpp [hunk 4]
```diff
@@ -13,13 +13,56 @@ using json = nlohmann::json;
...
     string file = "./tests/redis_multi_db_ut_config/database_config.json";
+    string nonexist_file = "./tests/redis_multi_db_ut_config/database_config_noexist.json";
+
     // by default , init should be false
     cout<<"Default : isInit = "<<SonicDBConfig::isInit()<<endl;
-    EXPECT_EQ(SonicDBConfig::isInit(), false);
+    EXPECT_FALSE(SonicDBConfig::isInit());
+
+
+    // load nonesist config file, should throw exception with file NO exist
```
Comment (qiluo-msft) [inline tests/redis_multi_db_ut.cpp]: >nonesist [](start = 12, length = 8)

nonexisting_file

Comment (dzhangalibaba) [reply tests/redis_multi_db_ut.cpp]: DONE

**Comment by dzhangalibaba** [comment]
> Minor issue.

FIXED

**Pull request merged**